salt/crypt.py

```diff
@@ -146,12 +146,12 @@ def gen_keys(keydir, keyname, keysize, user=None, passphrase=None):
                     cipher='des_ede3_cbc',
                     callback=lambda x: salt.utils.stringutils.to_bytes(passphrase))
         else:
-            with salt.utils.files.fopen(priv, 'wb') as f:
+            with salt.utils.files.fopen(priv, 'wb+') as f:
                 f.write(gen.exportKey('PEM', passphrase))
     if HAS_M2:
         gen.save_pub_key(pub)
     else:
-        with salt.utils.files.fopen(pub, 'wb') as f:
+        with salt.utils.files.fopen(pub, 'wb+') as f:
```

Comment:
Why was the `+` needed here? We're not reading from this filehandle.

Reply:
@terminalmage 'w+' mode truncates the file so here we 100% sure it will be written from the beginning. And the second: that mode just was there. I haven't found a commit where it was dropped. I don't see why this merge request has this change. So I've just reverted it back.
Am I wrong here?

Reply:
@DmitryKuzmenko According to the Python docs [here](https://docs.python.org/2/library/functions.html#open), both `w` and `w+` truncate the file. The only thing that adding the `+` does is open the filehandle in read-and-write mode, essentially adding funcs like `seek` to the filehandle object. Since we're not reading from the file here, it seems to me that `wb` is sufficient.

Reply:
Actually, I looked at the head of 2018.3 and it has the `+` already so I'm not sure how it was removed. I still don't think it's necessary, but it's not worth standing in the way of merging this.

Reply:
I'm going to merge-this as-is since it's a merge-forward. If it needs to be changed outside of this PR, it can be discussed in a new one. :)